libs/community/langchain_community/vectorstores/pinecone.py

```diff
@@ -359,11 +370,24 @@ def get_pinecone_index(
                 "Please install it with `pip install pinecone-client`."
             )
 
-        indexes = pinecone.list_indexes()  # checks if provided index exists
+        pinecone_client_version = get_distribution("pinecone-client").version
```

Comment:
would it be better to rely on `pinecone.__version__` instead? I think it isn't in the v3 release candidates, but might be good to add for consistency?

I believe `pkg_resources` is deprecated.

Reply:
@efriis we're getting `__version__` added back into `pinecone-client` — will get back to you once implemented

Reply:
Amazing. Thanks! Think that'll be better :)